apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts

```diff
@@ -220,4 +243,70 @@ export class PrepareAndValidateContentUsecase {
 
     return result;
   }
+
+  private async computeTierIssues(
+    defaultControlValues: Record<string, unknown>,
+    user: UserSessionData,
+    stepType?: StepTypeEnum
+  ): Promise<Record<string, ContentIssue[]>> {
+    const deferDuration =
+      isValidDigestUnit(defaultControlValues.unit) && isNumber(defaultControlValues.amount)
+        ? calculateMilliseconds(defaultControlValues.amount, defaultControlValues.unit)
+        : 0;
+
+    const restrictionsErrors = await this.tierRestrictionsValidatorUsecase.execute({
+      deferDurationMs: deferDuration,
+      organizationId: user.organizationId,
+      stepType,
+    });
+
+    if (!restrictionsErrors) {
+      return {};
+    }
+
+    const result: Record<string, ContentIssue[]> = {};
+    for (const restrictionsError of restrictionsErrors) {
+      result.amount = [
+        {
+          issueType: StepContentIssueEnum.TIER_LIMIT_EXCEEDED,
+          message: restrictionsError.message,
+        },
+      ];
+      result.unit = [
+        {
+          issueType: StepContentIssueEnum.TIER_LIMIT_EXCEEDED,
+          message: restrictionsError.message,
+        },
+      ];
+    }
```

Comment:
is the text what we want inside the usecase? maybe keep the usecase result structured, like max_allowed amount , max_allowed_unit and a pass/ fail for it, I would also maybe encapsulate it so it support move validations of similar nature in the future, so like have a nestring with an enum called DEFFER_DURATION

Reply:
we can encapsulate and refactor it to support other cases when we will have them, for not it will make the code more complex with no reason.

Reply:
my issue is that the usecase is not being really shared, and is just thorwing an exception on the worker no? no point for all the text there

Reply:
worker throwing an exception if there is an error, not the new usecase.